packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -221,101 +230,124 @@ export default class WorkerDeploy extends EasCommand {
           manifest,
         })
       );
-    } catch (error: any) {
-      progress.fail('Failed to prepare worker upload');
-      throw error;
-    }
-    progress.succeed('Prepared worker upload');
 
-    progress = ora('Creating worker deployment');
-    let deployResult: any;
-    try {
+      progress.text = 'Creating deployment';
       deployResult = await uploadTarballAsync(tarPath);
+      progress.succeed('Created deployment');
     } catch (error: any) {
-      progress.fail('Failed to create worker deployment');
+      progress.fail('Failed to create deployment');
       throw error;
     }
-    progress.succeed('Created worker deployment');
 
     await uploadAssetsAsync(assetMap, deployResult.uploads);
 
+    let deploymentAliasUrl: string | null = null;
     if (flags.aliasName) {
-      progress = ora(chalk`Assigning alias {bold ${flags.aliasName}} to worker deployment`).start();
+      progress = ora(chalk`Assigning alias {bold ${flags.aliasName}} to deployment`).start();
       try {
-        await assignWorkerDeploymentAliasAsync({
+        const workerAlias = await assignWorkerDeploymentAliasAsync({
           graphqlClient,
           appId: projectId,
           deploymentId: deployResult.id,
           aliasName: flags.aliasName,
         });
-        progress.succeed(chalk`Assigned alias {bold ${flags.aliasName}} to worker deployment`);
+        deploymentAliasUrl = workerAlias.url;
+
+        // Only stop the spinner when not promoting to production
+        if (!flags.isProduction) {
+          progress.succeed(chalk`Assigned alias {bold ${flags.aliasName}} to deployment`);
+        }
       } catch (error: any) {
-        progress.fail(chalk`Failed to assign {bold ${flags.aliasName}} alias to worker deployment`);
+        progress.fail(chalk`Failed to assign {bold ${flags.aliasName}} alias to deployment`);
         throw error;
       }
     }
 
-    const expoBaseDomain = process.env.EXPO_STAGING ? 'staging.expo' : 'expo';
-    const dashboardUrl = `https://${expoBaseDomain}.dev/projects/${projectId}/serverless/deployments`;
+    let deploymentProductionUrl: string | null = null;
+    if (flags.isProduction) {
+      try {
+        if (!flags.aliasName) {
+          progress = ora('Promoting deployment to production').start();
```

Comment:
isn't progress always set to be a spinner instance? when can this be executed?

Reply:
We want to combine both the "assigning alias" and "promote to production" spinners, to lower our usage of spinners to maximum of 3 checkmarks.

I agree it's a bit weird, but basically you should read this as:
  - If we aren't assigning an alias, create a new spinner
  - If we _do_ assign an alias, reuse the spinner from the alias assignment

Reply:
@szdziedzic sorry, I missed your video recording (did post some images of the output in the description). But here is the video:


https://github.com/user-attachments/assets/e302ae73-7467-4a09-b95c-73a80644236c